staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -946,10 +946,17 @@ func decode(codec runtime.Codec, versioner storage.Versioner, value []byte, objP
 	if _, err := conversion.EnforcePtr(objPtr); err != nil {
 		return fmt.Errorf("unable to convert output object to pointer: %v", err)
 	}
-	_, _, err := codec.Decode(value, nil, objPtr)
+	_, gvk, err := codec.Decode(value, nil, objPtr)
 	if err != nil {
+		if runtime.IsNotRegisteredError(err) && gvk != nil {
```

Comment:
does this ever return a non-nil gvk in an error condition?

Reply:
Now the code does not return non-nil gvk in an error conditiion, here just in case for future changes.